xlators/cluster/afr/src/afr-self-heal-metadata.c

```diff
@@ -343,7 +345,32 @@ __afr_selfheal_metadata_finalize_source(call_frame_t *frame, xlator_t *this,
     for (i = 0; i < priv->child_count; i++) {
         if (!sources[i] || i == source)
             continue;
-        if (!afr_xattrs_are_equal(replies[source].xdata, replies[i].xdata)) {
+        if (AFR_IS_ARBITER_BRICK(priv, i)) {
+            /* Source cannot be arbiter, hence we are not checking for it */
+            if (replies[i].xdata)
+                arbiter_dict = dict_copy_with_ref(replies[i].xdata, NULL);
+            if (replies[source])
+                src_dict = dict_ref(replies[source].xdata);
+
+            if (arbiter_brick || src_brick) {
```

Comment:
nit - shouldn't this part come before lines 350-353? What's the point of copying and ref'ing the dicts if we end up undoing this because of this statement?

Reply:
@mykaul Yes, I agree. I will change this.